test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -3684,6 +3684,11 @@ TEST_P(SslSocketTest, ProtocolVersions) {
   envoy::extensions::transport_sockets::tls::v3::TlsParameters* client_params =
       client.mutable_common_tls_context()->mutable_tls_params();
 
+  // Note: There aren't any valid TLSv1.0 or TLSv1.1 cipher suites enabled by default,
+  // so enable them to avoid false positives.
```

Comment:
Sorry what do you mean false positives? Do you just mean the test requires the use of non-default ciphers? Should we actually verify somehow that we don't negotiate with the new removed ciphers?

/wait

Reply:
This particular test (`ProtocolVersions`) verifies that we always negotiate the correct protocol version, and it enables/disables the non-default protocol versions in order to cover all variants.

However, now that we disabled all deprecated cipher suites, enabling `TLSv1.0` or `TLSv1.1` is no longer sufficient, since none of the new default cipher suites work for those older protocol versions, so the list of available cipher suites for them is effectively empty, and we need to enable older cipher suites in order to successfully negotiate those older protocol versions.

By "false positives" I mean that enabling those cipher suites let's us verify that those older protocol versions are rejected because of the protocol version negotiation, and not because no cipher suites can be negotiated.

I can add few cases to verify that we don't negotiate removed ciphers when using default config in a bit.

Reply:
Added, PTAL

Reply:
OK I see. Would it be better to split this out into multiple tests, one for each protocol version? It's kind of confusing that this is all mixed together? WDYT?

/wait-any

Reply:
There are 16 cases in the `ProtocolVersions` test that are well commented and very readable, IMHO:
https://github.com/envoyproxy/envoy/blob/092223355bab563e5d647f091de7eb96f9eeadf8/test/extensions/transport_sockets/tls/ssl_socket_test.cc#L3692-L3864

I don't think that splitting this test into 16 tiny tests with a ton of boilerplate would improve readability. We'd also need to do the same for other tests that follow this pattern.

If your issue is with the older cipher suite being enabled for all 16 tests cases ahead of time, then I could potentially set and unset them in each case that enables TLSv1.0 or TLSv1.1 and needs it, but that's more error-prone, and since we don't test negotiated cipher suites in this test, I'm not sure it's worth the noise.

Reply:
> If your issue is with the older cipher suite being enabled for all 16 tests cases ahead of time, then I could potentially set and unset them in each case that enables TLSv1.0 or TLSv1.1 and needs it, but that's more error-prone, and since we don't test negotiated cipher suites in this test, I'm not sure it's worth the noise.

Yes this was my concern. I trust your judgement on this so if you think this is best fine with me! Thank you for looking.